common/bolt11.c

```diff
@@ -431,6 +432,59 @@ static char *decode_r(struct bolt11 *b11,
         return NULL;
 }
 
+static void shift_bitmap_down(u8 *bitmap, size_t bits)
+{
+	u8 prev = 0;
+	assert(bits < CHAR_BIT);
+
+	for (size_t i = 0; i < tal_bytelen(bitmap); i++) {
+		/* Save top bits for next one */
+		u8 v = bitmap[i];
+		bitmap[i] = (prev | (v >> bits));
+		prev = (v << (8 - bits));
+	}
+	assert(prev == 0);
+}
+
+/* BOLT-a76d61dc9893eec75b2e9c4a361354c356c46894 #11:
+ *
+ * `9` (5): `data_length` variable. One or more bytes containing features
+ *  supported or required for receiving this payment.
+ *  See [Feature Bits](#feature-bits).
+ */
+static char *decode_9(struct bolt11 *b11,
+                      struct hash_u5 *hu5,
+                      u5 **data, size_t *data_len,
+                      size_t data_length)
+{
+        size_t flen = (data_length * 5 + 7) / 8;
+
+        b11->features = tal_arr(b11, u8, flen);
+        pull_bits_certain(hu5, data, data_len, b11->features,
+			  data_length * 5, true);
```

Comment:
spacing here seems off?

Reply:
It's actually fine, ut's just a tabs-vs-spaces thing.  Weird.

Reply:
@ZmnSCPxj explained it elsewhere as being from a tabs/spaces display problem w/ github (because of the additional '+' at the line start)